Source/DafnyLanguageServer.Test/Diagnostics/SimpleLinearVerificationDiagnosticTester.cs

```diff
@@ -0,0 +1,110 @@
+
+  [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+  public async Task EnsureVerificationDiagnosticsAreWorking() {
+    await VerifyTrace(@"
+ .  |  |  |  I  I  |  | :predicate Ok() {
+ .  |  |  |  I  I  |  | :  true
+ .  |  |  |  I  I  |  | :}
+    |  |  |  I  I  |  | :
+ .  S [S][ ][I][I][S] | :method Test(x: bool) returns (i: int)
+ .  S [=][=][-][-][~] | :   ensures i == 2
+ .  S [S][ ][I][I][S] | :{
+ .  S [S][ ][I][I][S] | :  if x {
+ .  S [S][ ][I][I][S] | :    i := 2;
+ .  S [=][=][-][-][~] | :  } else {
+ .  S [S][ ]/!\[I][S] | :    i := 1; //Next1:   i := /; //Next2:    i := 2;
+ .  S [S][ ][I][I][S] | :  }
+ .  S [S][ ][I][I][S] | :}
+    |  |  |  I  I  |  | :    
+ .  |  |  |  I  I  |  | :predicate OkBis() {
+ .  |  |  |  I  I  |  | :  false
+ .  |  |  |  I  I  |  | :}");
+  }
+  [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+  public async Task EnsuresItWorksForSubsetTypes() {
+    await VerifyTrace(@"
+    |  |  |  I  I  |  |  |  I  I  |  |  | :
+ .  |  |  |  I  I  |  |  |  I  I  |  |  | :ghost const maxId := 200;
+    |  |  |  I  I  |  |  |  I  I  |  |  | :
+ .  |  |  |  I  I  |  |  |  I  I  |  |  | :predicate isIssueIdValid(issueId: int) {
+ .  |  |  |  I  I  |  |  |  I  I  |  |  | :  101 <= issueId < maxId
+ .  |  |  |  I  I  |  |  |  I  I  |  |  | :}
+    |  |  |  I  I  |  |  |  I  I  |  |  | :
+ .  S  S  |  I  .  S  S [=] I  .  S  S  | :type IssueId = i : int | isIssueIdValid(i)
+ .  S  |  |  I  .  S  | [=] I  .  S  |  | :  witness 101 //Next1:   witness 99 //Next2:   witness 101 ");
+  }
+
+  [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+  public async Task EnsureItWorksForPostconditionsRelatedOutside() {
+    await VerifyTrace(@"
+ .  |  |  | :predicate F(i: int) {
+ .  |  |  | :  false // Should not be highlighted in gutter.
+ .  |  |  | :}
+    |  |  | :
+ .  S [S][ ]:method H()
+ .  S [=][=]:  ensures F(1)
+ .  S [=][=]:{
+ .  S [S][ ]:}");
+  }
+
+  [TestMethod, Timeout(MaxTestExecutionTimeMs * 10)]
+  public async Task EnsureNoAssertShowsVerified() {
+    for (var i = 0; i < 10; i++) {
+      await VerifyTrace(@"
+ .  |  |  |  I  | :predicate P(x: int)
+    |  |  |  I  | :
+ .  S [S][ ][I] | :method Main() {
+ .  S [=][=][I] | :  ghost var x :| P(x); //Next:  ghost var x := 1;
+ .  S [S][ ][I] | :}
+                | :");
+    }
+  }
+
+  [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+  public async Task EnsureEmptyDocumentIsVerified() {
+    await VerifyTrace(@"
+ | :class A {
+ | :}
+ | :");
+  }
+
+
+  [TestMethod/*, Timeout(MaxTestExecutionTimeMs)*/]
+  public async Task EnsuresEmptyDocumentWithParseErrorShowsError() {
+    await VerifyTrace(@"
+/!\:class A {/
+   :}
+   :");
+  }
+
+
+  [TestMethod/*, Timeout(MaxTestExecutionTimeMs)*/]
+  public async Task TopLevelConstantsHaveToBeVerifiedAlso() {
+    await VerifyTrace(@"
+    |  |  | :// The following should trigger only one error
+ .  |  |  | :ghost const a := [1, 2];
+    |  |  | :
+ .  S [~][=]:ghost const b := a[-1];");
```

Comment:
For the same reason as https://github.com/dafny-lang/dafny/pull/1942#discussion_r848477354, 
First, the error is found, so the line displays "error" (in error context). `[=]`, but since the implementation is not done verifying, it show `[~]`, although arguably there cannot be any more feedback there (except if there were another assertion verifying)
In practice, the `[~]` is barely visible.